nautobot/dcim/migrations/0030_migrate_region_and_site_data_to_locations.py

```diff
@@ -149,322 +170,321 @@ def create_site_location_type_locations(
+    JobHook = apps.get_model("extras", "jobhook")
+    Note = apps.get_model("extras", "note")
+    ObjectChange = apps.get_model("extras", "objectchange")
+    Relationship = apps.get_model("extras", "relationship")
+    RelationshipAssociation = apps.get_model("extras", "relationshipassociation")
+    WebHook = apps.get_model("extras", "webhook")
+
+    computed_fields = ComputedField.objects.filter(content_type=model_ct)
+    for cf in computed_fields:
+        cf.content_type = location_ct
+    ComputedField.objects.bulk_update(computed_fields, ["content_type"], 1000)
+
+    if model == "region":
+        ccs = ConfigContext.objects.filter(regions__isnull=False).prefetch_related("locations", f"{model}s")
+    else:
+        ccs = ConfigContext.objects.filter(sites__isnull=False).prefetch_related("locations", f"{model}s")
+    for cc in ccs:
+        if model == "region":
+            model_name_list = list(cc.regions.all().values_list("name", flat=True))
+        else:
+            model_name_list = list(cc.sites.all().values_list("name", flat=True))
+
+        model_locs = list(Location.objects.filter(name__in=model_name_list, location_type=model_lt))
+        if len(model_locs) < len(model_name_list):
+            logger.warning(
+                f'There is a mismatch between the number of {model_lt.name}s ({len(model_name_list)}) and the number of "{model_lt.name}" LocationType locations ({len(model_locs)})'
+                f" found in this ConfigContext {cc.name}"
+            )
+        cc.locations.add(*model_locs)
+        cc.save()
+
+    custom_fields = CustomField.objects.filter(content_types__in=[model_ct])
+    for cf in custom_fields:
+        cf.content_types.add(location_ct)
+
+    model_locs = Location.objects.filter(location_type=model_lt)
+    for model_loc in model_locs:
+        model_loc._custom_field_data = model_class.objects.get(migrated_location=model_loc)._custom_field_data
+    Location.objects.bulk_update(model_locs, ["_custom_field_data"], 1000)
+
+    # Custom Links and Image Attachements do not have Region ContentType as one of its ContentType options
+
+    dynamic_groups = DynamicGroup.objects.all()
+    for dg in dynamic_groups:
+        if f"{model}" in dg.filter:
+            dg.filter.setdefault("location", []).extend(dg.filter.pop(f"{model}"))
+        dg.save()
+
+    export_templates = ExportTemplate.objects.filter(content_type=model_ct)
+    for et in export_templates:
+        et.content_type = location_ct
+    ExportTemplate.objects.bulk_update(export_templates, ["content_type"], 1000)
+
+    job_hooks = JobHook.objects.filter(content_types__in=[model_ct])
+    for jh in job_hooks:
+        jh.content_types.add(location_ct)
+
+    notes = Note.objects.filter(assigned_object_type=model_ct)
+    for note in notes:
+        note.assigned_object_type = location_ct
+        old_model = model_class.objects.get(id=note.assigned_object_id)
+        note.assigned_object_id = old_model.migrated_location.id
```

Comment:
Same thought here and below about maybe not needing to update these object_id values?